### PR TITLE
[Version Control] Checkout problem using spaces at the end of the URL

### PR DESCRIPTION
**Pull request by jsuarezruiz**
Fix checkout error trying to use an url with spaces at the end

Fixes VSTS #901437

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepositoryEditor.cs [hunk 1]
```diff
@@ -122,14 +123,22 @@ protected virtual void OnRepositoryUrlEntryChanged(object sender, System.EventAr
...
 				updating = false;
 			}
 		}
-		
+
+		string SanitizerUrl(string url)
+		{
+			// Remove unnecessary spaces from the beginning or the end.
+			url = url.TrimStart ().TrimEnd ();
+
```
Comment (mrward) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepositoryEditor.cs]: Seems like we could use just `url.Trim ()` here.

Comment (Therzok) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepositoryEditor.cs]: Maybe do it in the Url setter?

Comment (jsuarezruiz) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepositoryEditor.cs]: I applied the suggestion of both. Thnx.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepositoryEditor.cs [hunk 2]
```diff
@@ -129,7 +129,7 @@ protected virtual void OnRepositoryUrlEntryChanged(object sender, System.EventAr
 				updating = false;
 			}
 		}
-
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepositoryEditor.cs]: Unrelated formatting change

Comment (jsuarezruiz) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepositoryEditor.cs]: Removed this change.

**Comment by jsuarezruiz** [comment]
@mrward Right. I have undone the unnecessary changes in `UrlBasedRepositoryEditor`.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepository.cs [hunk 3]
```diff
@@ -79,7 +79,7 @@ public virtual bool IsUrlValid (string url)
 		public virtual string Url
 		{
 			get { return url; }
-			set { url = value; CreateUri (); }
+			set { url = SanitizerUrl (value); CreateUri (); }
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepository.cs]: Potential NRE (CreateUri () has a null check), let's add a null check to `SanitizerUrl ()`, or – even better – just simplify this to:
```suggestion
			set { url = value?.Trim (); CreateUri (); }
```

Comment (jsuarezruiz) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UrlBasedRepository.cs]: Changed!

**Comment by sevoku** [comment]
@monojenkins backport release-8.2
